### PR TITLE
Cleanup of Maven code

### PR DESCRIPTION
**Pull request by cstamas**
undoing Preconditions messages, repeated String literals, misplaced annotations,
and some Sonar recommendations

http://bamboo.s/browse/NX3-OSSF422

### CODE REVIEW

File: plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/DigestExtractor.java [hunk 1]
```diff
@@ -116,14 +117,20 @@ public static String extract(final String input) {
...
   }
 
-  private static String compress(String digest) {
-    digest = digest.replaceAll(" ", "").toLowerCase(Locale.US);
-    return digest;
+  /**
+   * Removes white space and lower cases the digest string.
+   */
+  private static String compress(final String digest) {
+    return digest.replaceAll(" ", "").toLowerCase(Locale.US);
```
Comment (mcculls) [inline plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/DigestExtractor.java]: opportunity here to replace `toLowerCase` with `Strings2.lower` for consistency

**Comment by mcculls** [comment]
+1 good improvements

**Comment by jdillon** [comment]
+1

**Comment by adreghiciu** [comment]
+1

**Comment by cstamas** [comment]
857d2f618b013adf10c9530a2fad7c2b3b6c8237

**Pull request closed without merging**